### PR TITLE
[DOC] update notes in download page for the decompressing gzip issue [skip ci]

### PR DESCRIPTION
**Pull request by nvliyuan**
update notes in download page for the decompressing gzip issue 
and fix some dead links

### CODE REVIEW

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
please help review the doc update @sameerz @mattahrens @GaryShen2008 thanks

File: docs/download.md [hunk 1]
```diff
@@ -101,6 +101,9 @@ The output of signature verify:
 * Improve UCX shuffle 
 * For updates on RAPIDS Accelerator Tools, please visit [this link](https://github.com/NVIDIA/spark-rapids-tools/releases)
 
+Note: There is a known issue for 24.08.1 release while decompressing gzip files on H100 GPU.
```
Comment (sameerz) [inline docs/download.md:104]:
```suggestion
Note: There is a known issue in the 24.08.1 release when decompressing gzip files on H100 GPUs.  
```

**Review by sameerz — APPROVED** [review]

**Comment by sameerz** [comment]
build

**Review by GaryShen2008 — APPROVED** [review]

**Pull request merged**